### PR TITLE
VPN-6479 - handle native messaging

### PR DESCRIPTION
**Pull request by strseb**
This PR adds a `VPNController` to the Extension. 
-> It gets instaniated by main, and opens a Native Messaging port. 
-> One can send messages to the vpn using it's `sendToApp`

One can subscribe to the Controller using `.subscribe` in the Backend Script or `
const port  = globalThis.chrome.runtime.connect({
  name: "vpncontroller"
});` in the contentscript. 

Anytime somethings changes the Controller emits a new immutable object implementing `VPNState` - containing the activation status, serverlist and local socket proxy status.  

Also this add's tests for the state thing using jest. And a few nithings to make sure VsCode autocomplete work

### CODE REVIEW

File: src/background/requestHandler.js [hunk 1]
```diff
@@ -4,22 +4,33 @@
 
 import {Component} from "./component.js";
 import {Logger} from "./logger.js";
+import {VPNController,VPNState } from "./vpncontroller/index.js";
```
Comment (lesleyjanenorton) [inline src/background/requestHandler.js:7]:
```suggestion
import {VPNController, VPNState} from "./vpncontroller/index.js";
```
Nit!

**Review by lesleyjanenorton — APPROVED** [review]
Nice, Basti!! R+ wc. Happy to merge now and address later.

File: src/background/vpncontroller/states.js [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+ */
+export class StateVPNUnavailable extends VPNState {
+    state = "Unavailable";
+    installed = false;
+    connected = false;
+}
+
+/**
+ * This state is used if the VPN Client is 
+ * alive but the Connection is Disabled
```
[the remark this answers is not in the document]
Comment (strseb) [reply src/background/vpncontroller/states.js:93]: Ah yes! Sorry i've changed the state instead. 
That was from mac where it considered if the client was dead == !installed. 
But you know the client could crash, right click close and stuff 😅.

File: src/background/vpncontroller/states.js [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+    * 
+    * @param {VPNState} state - The state to replicate
+    * @param {browser.storage.StorageArea} storage - The storage area to look for
+    * @returns {Promise<VPNState>} - Returns a copy of the state, or the same in case of missing data.
+    */
+    static async fromStorage(state = new StateVPNUnavailable(null), storage = browser.storage.local) {
+        const { mozillaVpnServers } = await storage.get(MOZILLA_VPN_SERVERS_KEY);
+        if (typeof (mozillaVpnServers) === "undefined") {
+            await storage.set({ [MOZILLA_VPN_SERVERS_KEY]: [] });
+            await storage.set({ [MOZILLA_VPN_HIDDEN_TOUTS_LIST_KEY]: [] });
```
[the remark this answers is not in the document]
Comment (lesleyjanenorton) [reply src/background/vpncontroller/states.js]: Oh same here. And secure proxy. And fedifox.

**Pull request merged**